test/software/amazon/ion/EquivsTestCase.java

```diff
@@ -187,12 +195,40 @@ protected void checkEquivalence(final IonValue left,
         }
     }
 
+    public static IonDatagram[] roundTripDatagram(IonDatagram input) throws IOException{
+        IonSystem system = IonSystemBuilder.standard().build();
+        IonLoader loader = system.getLoader();
+        ByteArrayOutputStream textOutputStream = new ByteArrayOutputStream();
+        ByteArrayOutputStream binaryOutputStream = new ByteArrayOutputStream();
+        IonWriter textWriter = IonTextWriterBuilder.standard().build(textOutputStream);
+        IonWriter binaryWriter = IonBinaryWriterBuilder.standard().build(binaryOutputStream);
+        IonReader reader = system.newReader(input);
+        textWriter.writeValues(reader);
+        textWriter.close();
+        reader = system.newReader(input);
```

Comment:
The previous reader needs to be closed first, and this new reader must eventually be closed too. Consider putting the closing of both the reader and writer in a `finally` that follows a `try` that contains the lines of code that use them (namely `*Writer.writeValues(reader);`.